aws/resource_aws_sfn_state_machine.go

```diff
@@ -112,15 +120,14 @@ func resourceAwsSfnStateMachineRead(d *schema.ResourceData, meta interface{}) er
 	})
 	if err != nil {
 
-		if awserr, ok := err.(awserr.Error); ok {
-			if awserr.Code() == "NotFoundException" || awserr.Code() == "StateMachineDoesNotExist" {
-				d.SetId("")
-				return nil
-			}
+		if isAWSErr(err, sfn.ErrCodeStateMachineDoesNotExist, "") {
+			d.SetId("")
```

Comment:
Nit: We should return a warning log message here:

```suggestion
			log.Printf("[WARN] SFN State Machine (%s) not found, removing from state", d.Id())
			d.SetId("")
```